src/module-init/module-init.c

```diff
@@ -209,6 +210,15 @@ int RediSearch_Init(RedisModuleCtx *ctx, int mode) {
   GC_ThreadPoolStart();
 
   CleanPool_ThreadPoolStart();
+  
+  // Init threadpool. 
+  // Threadpool size can only be set on load, hence it is not dependent on
+  // threadsEnabled flag.
```

Comment:
isn't it should be?

Reply:
if WORKERS_THREAD is set, we create a threadpool, even if ENABLE_THREADS is off because ENABLE_THREADS can be later set on, but  WORKERS_THREAD should never change.